### ISSUE TITLE
Problems related with latest Pandas library (2.2.3)

### ISSUE THREAD

**Issue by rlgus94** [sascha-kirch/samba-mixer#5]
Good sir, I appreciate your inspiring work.

I wanted to report a few issues I encountered while working with the latest version of pandas (2.2.3 as of 11/13/2024).
(which is automatically installed via pyproject.toml)

1. The latest pandas update (2.2.3) appears to have issues with DataFrame creation, which is affecting the preprocessing scripts. This problem is addressed in [GitHub issue #60184](https://github.com/pandas-dev/pandas/issues/60184) and can be resolved by downgrading pandas to version 2.2.x. In my case, 2.2.1 and 2.2.2 both works.

2. In samba_mixer/dataset/preprocessing.py, I noticed that int64 type values in the 'battery_id' and 'cycle_id' columns are causing a TypeError. This issue can be resolved by casting these values to int32, as shown below:

![image](https://github.com/user-attachments/assets/1cd3b2e2-7d36-4e91-9e4a-a415e587f6f9)

This issue can be solved by casting these values to int32.

```python
def filter_k_first_cycles(
    df: pd.DataFrame,
    test_battery_cycle_start: Dict[int, int],
) -> pd.DataFrame:
    """Filters cycles according to the provided `test_battery_cycle_start`.

    Args:
        df (pd.DataFrame): BAttery Data
        test_battery_cycle_start (Dict[int,int]): Dict with keys beeing battery_ids and values start cycles.

    Returns:
        pd.DataFrame: Filtered DataFrame
    """
    # df columns:
    # Index(['battery_id', 'cycle_id', 'time_cycle_start', 'ambient_temperature',
    #    'num_samples', 'capacity_k', 'capacity_0', 'fade', 'cutoff_voltage',
    #    'discharge_type', 'discharge_amplitude', 'discharge_frequency',
    #    'discharge_dutycycle', 'soh', 'data_file', 'time_diff_hours', 'split'],
    #   dtype='object')
    # int64 type causes issues. force to int32 to solve this issue.
    df['battery_id'] = df['battery_id'].astype("Int32")
    df['cycle_id'] = df['cycle_id'].astype("Int32")

    for battery_id_to_filter, start_cycle in test_battery_cycle_start.items():
        df = df.query(f" \
                (battery_id == {battery_id_to_filter} and cycle_id >= {start_cycle})\
                or (battery_id != {battery_id_to_filter})")

        df.loc[df["battery_id"] == battery_id_to_filter,
               "cycle_id"] -= start_cycle
    return df
    ```
    
    Thank you for your hard work. Hope this helps.

**Comment by sascha-kirch** [comment]
Hello @rlgus94, 

first of all thank you very much for reporting the issue and putting so much effort into the description. Really appreciate it! 
I just doubled checked my python packages and I have `pandas 1.3.5`. 

Could it be that you did not use the docker container, but installed the repo into your own environment? 
In our docker container we use an [NVIDIA image](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel_22-04.html#rel_22-04) that already contains a pandas installation, hence when installing samba-mixer the requirement of pandas would already be satisfied. And since we do not specify the version in our requirements, it will install the latest version of pandas which is 2.2.3 at the moment. 

If you need some support with the docker container, we might be able to help you so you are not blocked! 

Best,
Sascha

**Comment by sascha-kirch** [comment]
Here the output of `pip list` inside my container as refference:
[samba_mixer_packages.txt](https://github.com/user-attachments/files/17746421/samba_mixer_packages.txt)

**Comment by rlgus94** [comment]
Hello @sascha-kirch,

Thank you for your valuable time.

I have confirmed that creating the Docker container with the provided Dockerfile resolves the issue, as you suggested.

Pandas version 1.3.5 is installed, and the entire project operates flawlessly.

I have already executed S4D and S6 models within the Docker container I utilize, anticipating no difficulties. However, it appears that the most recent versions of NumPy and Pandas are causing several issues.

Thank you sincerely for your assistance.
Have a pleasant day!
